src/main/java/org/prebid/server/bidder/adverxo/AdverxoBidder.java

```diff
@@ -0,0 +1,195 @@
+import org.prebid.server.proto.openrtb.ext.request.adverxo.ExtImpAdverxo;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AdverxoBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpAdverxo>> ADVERXO_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+    private static final String ADUNIT_MACROS_ENDPOINT = "{{adUnitId}}";
+    private static final String AUTH_MACROS_ENDPOINT = "{{auth}}";
+    private static final String PRICE_MACRO = "${AUCTION_PRICE}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    public AdverxoBidder(String endpointUrl, JacksonMapper mapper,
+                         CurrencyConversionService currencyConversionService) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = mapper;
+        this.currencyConversionService = currencyConversionService;
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            try {
+                final ExtImpAdverxo extImp = parseImpExt(imp);
+                final String endpoint = resolveEndpoint(extImp);
+                final Imp modifiedImp = modifyImp(request, imp);
+                final BidRequest outgoingRequest = createRequest(request, modifiedImp);
+
+                requests.add(BidderUtil.defaultRequest(outgoingRequest, endpoint, mapper));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(requests, errors);
+    }
+
+    private ExtImpAdverxo parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), ADVERXO_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Error parsing ext.imp.bidder: " + e.getMessage());
+        }
+    }
+
+    private String resolveEndpoint(ExtImpAdverxo extImp) {
+        return endpointUrl
+                .replace(ADUNIT_MACROS_ENDPOINT,
+                        extImp.getAdUnitId() == null ? StringUtils.EMPTY : String.valueOf(extImp.getAdUnitId()))
+                .replace(AUTH_MACROS_ENDPOINT, HttpUtil.encodeUrl(StringUtils.defaultString(extImp.getAuth())));
+    }
```

Comment:
Default to 0 instead of empty string
```
    private String resolveEndpoint(ExtImpAdverxo extImp) {
         final String adUnitId = extImp.getAdUnitId();
         return endpointUrl
                 .replace(ADUNIT_MACROS_ENDPOINT, adUnitId == null ? 0 : String.valueOf(adUnitId))
                 .replace(AUTH_MACROS_ENDPOINT, HttpUtil.encodeUrl(StringUtils.defaultString(extImp.getAuth())));
     }
```